File: java/src/solutions/3731_find_missing_elements.java
import java.util.ArrayList;
import java.util.List;

class Solution {
    public List<Integer> findMissingElements(int[] nums) {
        boolean[] bucket = new boolean[101];
        int min = 101;
        int max = 0;
        for (int n : nums) {
            bucket[n] = true;
            min = Math.min(min, n);
            max = Math.max(max, n);
        }
        List<Integer> result = new ArrayList<>();
        for (int i = min + 1; i < max; i++){
            if (!bucket[i]){
                result.add(i);
            }
        }
        return result;
    }
}

File: rust/src/bin/3731_find_missing_elements.rs
impl Solution {
    pub fn find_missing_elements(nums: Vec<i32>) -> Vec<i32> {
        let mut bucket = [false; 101];
        let (mut min, mut max) = (101, 0);
        for n in nums {
            bucket[n as usize] = true;
            max = max.max(n);
            min = min.min(n);
        }
        (min..max).filter(|&x| !bucket[x as usize]).collect()
    }
}

struct Solution {}
